src/OpenDebugAD7/TextPositionTuple.cs

```diff
@@ -24,30 +36,132 @@ private TextPositionTuple(Source source, int line, int column)
         public static TextPositionTuple GetTextPositionOfFrame(PathConverter converter, IDebugStackFrame2 frame)
         {
             IDebugDocumentContext2 documentContext;
+            var source = new Source();
             if (frame.GetDocumentContext(out documentContext) == 0 &&
                 documentContext != null)
             {
-                TEXT_POSITION[] beginPosition = new TEXT_POSITION[1];
-                TEXT_POSITION[] endPosition = new TEXT_POSITION[1];
+                var beginPosition = new TEXT_POSITION[1];
+                var endPosition   = new TEXT_POSITION[1];
                 documentContext.GetStatementRange(beginPosition, endPosition);
 
                 string filePath;
                 documentContext.GetName(enum_GETNAME_TYPE.GN_FILENAME, out filePath);
 
                 string convertedFilePath = converter.ConvertDebuggerPathToClient(filePath);
+                // save the filename as we need it in any case
+                source.Name = Path.GetFileName(convertedFilePath);
 
-                Source source = new Source()
+                // we have debug info but do we actually have the source code?
+                if (File.Exists(convertedFilePath))
                 {
-                    Path = convertedFilePath,
-                    Name = Path.GetFileName(convertedFilePath)
-                };
-                int line = converter.ConvertDebuggerLineToClient((int)beginPosition[0].dwLine);
-                int column = unchecked((int)(beginPosition[0].dwColumn + 1));
+                    source.Path = convertedFilePath;
+                    int srcline = converter.ConvertDebuggerLineToClient((int)beginPosition[0].dwLine);
+                    int srccolumn = unchecked((int)(beginPosition[0].dwColumn + 1));
 
-                return new TextPositionTuple(source, line, column);
+                    return new TextPositionTuple(source, srcline, srccolumn);
+                }
             }
```

Comment:
>} [](start = 12, length = 1)

A few comments for the rest of this:
1. We need a client check - we want to do this stuff with an IDE that has real disassembly support
2. I don't think you want to do the disassembly up front. Instead we should keep a collection of sourceReferences and do the disassembly when the IDE requests source.

Reply:
Makes sense.
The thing is there seems to be no way to find the start of the current function in gdb.
So different address requests belonging to the same function without actually disassembling seem to be impossible to relate.

Reply:
Looks like VS uses DisassembleRequest though?
https://github.com/microsoft/MIEngine/commit/b2a5c7a891e949f129d8ded15aa320cc6409f910

Reply:
Yes, in the VS case, we want to return whatever source path we have in the symbols. VS is fine with source paths that don't exist and will request disassembly if that is what the user chooses to do.

Reply:
But how to handle this? I guess there'd need to be a client support flag to do it properly.

Reply:
Do you mean how to check for VS? If yes, Gabrielle added support for this to AD7Session recently: https://github.com/gregg-miskelly/MIEngine/blob/896c72d633b39a0571b9599d15c5a8fd0f1fcf58/src/OpenDebugAD7/AD7DebugSession.cs#L632-L638